lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceEvent.java

```diff
@@ -58,4 +60,11 @@
    * @return Runtime name of the resource
    */
   String getRuntimeName();
+
+  /**
+   * @return Node labels of the resource.
+   * In YARN, key is a static constant string and value is the node label.
+   * In Mesos, these are required Attributes.
+   */
+  Map<String, String> getNodeLabels();
```

Comment:
Please sort the comment such that the `@return` is at the bottom.

This is interesting. Maybe the correct name for this field isn't "Node labels", but "Scheduling Constraints" instead? If we have this `string,string` map, we could funnel all sorts of constraints through to the various runtimes, right?